corehq/apps/sms/tasks.py

```diff
@@ -48,6 +49,8 @@ def handle_unsuccessful_processing_attempt(msg):
     msg.num_processing_attempts += 1
     if msg.num_processing_attempts < settings.SMS_QUEUE_MAX_PROCESSING_ATTEMPTS:
         delay_processing(msg, settings.SMS_QUEUE_REPROCESS_INTERVAL)
+    elif RETRY_SMS_INDEFINITELY.enabled(msg.domain):
```

Comment:
could we also add a check here that msg.direction == OUTGOING? just to be explicit that this won't apply to inbound messages